ProjFS.Mac/PrjFSKext/PrjFSKext/VirtualizationRoots.cpp

```diff
@@ -37,9 +93,16 @@ kern_return_t VirtualizationRoots_Init()
         return KERN_FAILURE;
     }
     
-    for (uint32_t i = 0; i < MaxVirtualizationRoots; ++i)
+    s_maxVirtualizationRoots = 128;
+    s_virtualizationRoots = Memory_AllocArray<VirtualizationRoot>(s_maxVirtualizationRoots);
+    if (nullptr == s_virtualizationRoots)
+    {
+        return KERN_RESOURCE_SHORTAGE;
```

Comment:
Everywhere else we return KERN_FAILURE if we fail to allocate memory. Should we update those to return this error code instead?

Reply:
Given that the return value is only checked as a boolean, it doesn't really matter much I suppose. If I was designing it from scratch, I'd probably either pass the full return value back out from the kext start function, or simply return booleans from the various init functions. I don't have sufficiently strong feelings either way and can also change this to `KERN_FAILURE` if preferred.